### PR TITLE
Issue 1808

### PR DESCRIPTION
**Pull request by Bharath-Sathappan**
This change provides the ability to use dataframe index in `beta_diversity`, fixes issue #1808

Please complete the following checklist:

* [x] I have read the guidelines in [CONTRIBUTING.md](https://github.com/biocore/scikit-bio/blob/master/.github/CONTRIBUTING.md).

* [x] I have documented all public-facing changes in [CHANGELOG.md](https://github.com/biocore/scikit-bio/blob/master/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/biocore/scikit-bio/blob/master/LICENSE.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied. **It is your responsibility to disclose code, documentation, or other content derived from external source(s).** If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [x] **This pull request does not include code, documentation, or other content derived from external source(s).**

**Note:** [REVIEWING.md](https://github.com/biocore/scikit-bio/blob/master/REVIEWING.md) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 ### Features
 
+* `beta_diversity` now supports use of Pandas `DataFrame` index issue [#1808](https://github.com/scikit-bio/scikit-bio/issues/1808).
```
Comment (wasade) [inline CHANGELOG.md:11]:
```suggestion
* `beta_diversity` now supports use of Pandas a `DataFrame` index, issue [#1808](https://github.com/scikit-bio/scikit-bio/issues/1808).
```

File: skbio/diversity/_driver.py [hunk 2]
```diff
@@ -396,6 +396,9 @@ def beta_diversity(metric, counts, ids=None, validate=True, pairwise_func=None,
     sklearn.metrics.pairwise_distances
 
     """
+    if isinstance(counts, pd.core.frame.DataFrame) and ids is None:
```
Comment (wasade) [inline skbio/diversity/_driver.py:399]:
```suggestion
    if isinstance(counts, pd.DataFrame) and ids is None:
```

...I think that should work given:

```In [1]: import pandas as pd

In [2]: d = pd.DataFrame([])

In [3]: isinstance(d, pd.DataFrame)
Out[3]: True
```

**Review by wasade — CHANGES_REQUESTED** [review]
Just a few minor suggested changes

**Review by wasade — APPROVED** [review]

**Comment by qiyunzhu** [comment]
Good job! @Bharath-Sathappan @wasade 

In the future we may consider adopting the scikit-learn [way](https://github.com/scikit-learn/scikit-learn/blob/3f89022fa04d293152f1d32fbc2a5bdaaf2df364/sklearn/utils/validation.py#L2009) such that the functions can work with any compatible table object, including pd.Dataframe, biom.Table and more.

**Pull request merged**